### ISSUE TITLE
`no-node-access` to report where fireEvent/userEvent could be used

### ISSUE THREAD

**Issue by zaicevas** [testing-library/eslint-plugin-testing-library#752]
### What rule do you want to change?

no-node-access

### Does this change cause the rule to produce more or fewer warnings?

More warnings

### How will the change be implemented?

Not sure regarding the versioning (major/breaking change), but the idea behind this is that `userEvent` or `fireEvent` usage should be preferred over direct node access to call events, such as `.focus()`, `.click()`. We could have a whitelist of methods that would be reported by the rule.

### Example code

```tsx
screen.getByPlaceholderText('Search...').focus()
```

### How does the current rule affect the code?

No error reported

### How will the new rule affect the code?

Report an error

### Do you want to submit a pull request to change the rule?

No

**Comment by Belco90** [comment]
Hi @zaicevas. It sounds like an excellent idea, actually. Not sure if fits better in this rule, or in a new one. Still, would be nice to report this. Are you up to work on it?

**Comment by zaicevas** [comment]
I'll try to evaluate the complexity and let you know :)

**Comment by zaicevas** [comment]
Apologies for late answer, but I don't have capacity right now.

**Comment by Simek** [comment]
@Belco90 On 7.5.0 release this rule now errors on `userEvent` usage which seems to be a false-positive:

![Image](https://github.com/user-attachments/assets/dd1b17e2-1610-45f8-8021-bf6c315bbdb3)

**Comment by Belco90** [comment]
> On 7.5.0 release this rule now errors on userEvent usage which seems to be a false-positive

@Simek Thanks for reporting this. Can you create a new issue for it?

**Comment by Simek** [comment]
@Belco90 Sure! Reported as:
* #1024
